lane.py

```diff
@@ -387,897 +372,796 @@ def has_next_page(self):
+    library_id = Column(Integer, ForeignKey('libraries.id'), index=True,
+                        nullable=False)
+    parent_id = Column(Integer, ForeignKey('lanes.id'), index=True,
+                       nullable=True)
+
+    # A lane may have one parent lane and many sublanes.
+    sublanes = relationship(
+        "Lane", 
+        backref=backref("parent", remote_side = [id]),
+    )
 
-        if fiction is None:
-            fiction = cls.BOTH_FICTION_AND_NONFICTION
-        return genres, fiction
+    # A lane may have multiple associated LaneGenres. For most lanes,
+    # this is how the contents of the lanes are defined.
+    genres = association_proxy('lane_genres', 'genre',
+                               creator=LaneGenre.from_genre)
+    lane_genres = relationship(
+        "LaneGenre", foreign_keys="LaneGenre.lane_id", backref="lane"
+    )
 
-    def works(self, facets=None, pagination=None):
-        """Find Works that will go together in this Lane.
+    # identifier is a name for this lane that is unique across the
+    # library.  "Adult Fiction" is a good example. We can't have an
+    # adult fiction lane called "Fiction" and a YA fiction lane called
+    # "Fiction".
+    identifier = Column(Unicode)
+
+    # display_name is the name of the lane as shown to patrons.  It's
+    # okay for this to be duplicated across the library, but it's not
+    # okay to have two lanes with the same parent and the same display
+    # name -- that would be confusing.
+    display_name = Column(Unicode)
+
+    # True = Fiction only
+    # False = Nonfiction only
+    # null = Both fiction and nonfiction
+    #
+    # This may interact with lane_genres, for genres such as Humor
+    # which can apply to either fiction or nonfiction.
+    fiction = Column(Boolean, index=True, nullable=True)
+
+    # A lane may be restricted to works classified for specific audiences
+    # (e.g. only Young Adult works).
+    _audiences = Column(ARRAY(Unicode), name='audiences')
+
+    # A lane may further be restricted to works classified as suitable
+    # for a specific age range.
+    _target_age = Column(INT4RANGE, name="target_age", index=True)
+
+    # A lane may be restricted to works available in certain languages.
+    languages = Column(ARRAY(Unicode))
+
+    # A lane may be restricted to works in certain media (e.g. only
+    # audiobooks).
+    media = Column(ARRAY(Unicode))
+
+    # TODO: At some point it may be possible to restrict a lane to certain
+    # formats (e.g. only electronic materials or only codices).
+
+    # Only books licensed through this DataSource will be shown.
+    license_datasource_id = Column(
+        Integer, ForeignKey('datasources.id'), index=True,
+        nullable=True
+    )
 
-        Works will:
+    # Only books on one or more CustomLists obtained from this
+    # DataSource will be shown.
+    _list_datasource_id = Column(
+        Integer, ForeignKey('datasources.id'), index=True,
+        nullable=True
+    )
```

Comment:
What will this be used for?

Reply:
I'm guessing NYT, but that's one list so I'm wondering why it would be associated with the data source instead.

Reply:
NYT has about 30 different best-seller lists (paperback fiction, hardcover nonfiction, romance, etc.)